dashboard/src/actions/datasetListActions.js

```diff
@@ -1,27 +1,74 @@
 import * as TYPES from "./types";
 
 import API from "../utils/axiosInstance";
+import { DANGER } from "assets/constants/toastConstants";
+import { showToast } from "./toastActions";
 import { uriTemplate } from "utils/helper";
 
-export const fetchPublicDatasets = () => async (dispatch, getState) => {
+export const fetchPublicDatasets = (page) => async (dispatch, getState) => {
   try {
     dispatch({ type: TYPES.LOADING });
     const endpoints = getState().apiEndpoint.endpoints;
+    const { offset, limit, filter, searchKey, perPage } =
+      getState().datasetlist;
+    let publicData = [...getState().datasetlist.publicData];
+    const params = new URLSearchParams();
+    params.append("metadata", "dataset.uploaded");
+    params.append("access", "public");
+    params.append("offset", offset);
+    params.append("limit", limit);
+
+    if (searchKey) {
+      params.append("name", searchKey);
+    }
+    if (filter.startDate instanceof Date && !isNaN(filter.startDate)) {
+      params.append("start", new Date(filter.startDate).toUTCString());
+    }
+    if (filter.endDate instanceof Date && !isNaN(filter.endDate)) {
+      params.append("end", new Date(filter.endDate).toUTCString());
+    }
+
     const response = await API.get(
       uriTemplate(endpoints, "datasets_list", {}),
-      { params: { metadata: "dataset.uploaded", access: "public" } }
+      { params }
     );
+
     if (response.status === 200 && response.data) {
+      const startIdx = (page - 1) * perPage;
+
+      if (publicData.length === 0) {
+        publicData = new Array(response.data.total);
+      }
+      publicData.splice(
+        startIdx,
+        response.data.results.length,
+        ...response.data.results
+      );
+
+      dispatch({
+        type: TYPES.GET_PUBLIC_DATASETS,
+        payload: [...publicData],
+      });
+      // in case of last page, next_url is empty
+      if (response.data.next_url) {
+        const urlSearchParams = new URLSearchParams(response.data.next_url);
+        const params = Object.fromEntries(urlSearchParams.entries());
+
+        dispatch({
+          type: TYPES.SET_PAGE_OFFSET,
+          payload: Number(params?.offset),
+        });
```

Comment:
@MVarshini, don't we need an `else` clause, here?  That is, if we get to the last page, we still want to do a `SET_PAGE_OFFSET`, right?